enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj

```diff
@@ -44,302 +45,307 @@
 ;;; `(into [] (extract/extract ...))` in these tests.
 
 (deftest load-basics-test
-  (testing "a simple, fresh collection is imported"
-    (let [serialized (atom nil)
-          eid1       "0123456789abcdef_0123"]
-      (ts/with-dbs [source-db dest-db]
-        (testing "extraction succeeds"
-          (ts/with-db source-db
-            (ts/create! :model/Collection :name "Basic Collection" :entity_id eid1)
-            (reset! serialized (into [] (serdes.extract/extract {})))
-            (is (some (fn [{[{:keys [model id]}] :serdes/meta}]
-                        (and (= model "Collection") (= id eid1)))
-                      @serialized))))
-
-        (testing "loading into an empty database succeeds"
-          (ts/with-db dest-db
-            (serdes.load/load-metabase! (ingestion-in-memory @serialized))
-            ;; the trash is serialized and loaded, so restrict to nil type
-            (let [colls (t2/select :model/Collection :type nil)]
-              (is (= 1 (count colls)))
-              (is (= "Basic Collection" (:name (first colls))))
-              (is (= eid1               (:entity_id (first colls)))))))
+  (with-redefs [search/reindex! (constantly nil)]
```

Comment:
We need a linter for this, but tests should always use `mt/with-dynamic-fn-redefs` instead, which makes it thread safe.